org.eclipse.jdt.ls.core/src/org/eclipse/jdt/ls/core/internal/semantictokens/SemanticTokensVisitor.java

```diff
@@ -448,6 +456,67 @@ public boolean visit(RecordDeclaration node) {
 		return false;
 	}
 
+	@Override
+	public boolean visit(TypeDeclaration node) {
+		acceptNode(node.getJavadoc());
+		acceptNodeList(node.modifiers());
+
+		int typeKeywordStart = -1;
```

Comment:
I would just add a comment here explaining we're using the scanner because there's no nice way to get the type declaration key word through the AST model.